{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [],
   "source": [
    "import seaborn as sns\n",
    "df = sns.load_dataset('titanic')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>survived</th>\n",
       "      <th>pclass</th>\n",
       "      <th>sex</th>\n",
       "      <th>age</th>\n",
       "      <th>sibsp</th>\n",
       "      <th>parch</th>\n",
       "      <th>fare</th>\n",
       "      <th>embarked</th>\n",
       "      <th>class</th>\n",
       "      <th>who</th>\n",
       "      <th>adult_male</th>\n",
       "      <th>deck</th>\n",
       "      <th>embark_town</th>\n",
       "      <th>alive</th>\n",
       "      <th>alone</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>0</td>\n",
       "      <td>3</td>\n",
       "      <td>male</td>\n",
       "      <td>22.0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>7.2500</td>\n",
       "      <td>S</td>\n",
       "      <td>Third</td>\n",
       "      <td>man</td>\n",
       "      <td>True</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Southampton</td>\n",
       "      <td>no</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>female</td>\n",
       "      <td>38.0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>71.2833</td>\n",
       "      <td>C</td>\n",
       "      <td>First</td>\n",
       "      <td>woman</td>\n",
       "      <td>False</td>\n",
       "      <td>C</td>\n",
       "      <td>Cherbourg</td>\n",
       "      <td>yes</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1</td>\n",
       "      <td>3</td>\n",
       "      <td>female</td>\n",
       "      <td>26.0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>7.9250</td>\n",
       "      <td>S</td>\n",
       "      <td>Third</td>\n",
       "      <td>woman</td>\n",
       "      <td>False</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Southampton</td>\n",
       "      <td>yes</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>female</td>\n",
       "      <td>35.0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>53.1000</td>\n",
       "      <td>S</td>\n",
       "      <td>First</td>\n",
       "      <td>woman</td>\n",
       "      <td>False</td>\n",
       "      <td>C</td>\n",
       "      <td>Southampton</td>\n",
       "      <td>yes</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>0</td>\n",
       "      <td>3</td>\n",
       "      <td>male</td>\n",
       "      <td>35.0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>8.0500</td>\n",
       "      <td>S</td>\n",
       "      <td>Third</td>\n",
       "      <td>man</td>\n",
       "      <td>True</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Southampton</td>\n",
       "      <td>no</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   survived  pclass     sex   age  sibsp  parch     fare embarked  class  \\\n",
       "0         0       3    male  22.0      1      0   7.2500        S  Third   \n",
       "1         1       1  female  38.0      1      0  71.2833        C  First   \n",
       "2         1       3  female  26.0      0      0   7.9250        S  Third   \n",
       "3         1       1  female  35.0      1      0  53.1000        S  First   \n",
       "4         0       3    male  35.0      0      0   8.0500        S  Third   \n",
       "\n",
       "     who  adult_male deck  embark_town alive  alone  \n",
       "0    man        True  NaN  Southampton    no  False  \n",
       "1  woman       False    C    Cherbourg   yes  False  \n",
       "2  woman       False  NaN  Southampton   yes   True  \n",
       "3  woman       False    C  Southampton   yes  False  \n",
       "4    man        True  NaN  Southampton    no   True  "
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Requirement already satisfied: dtale in f:\\anaconda\\lib\\site-packages (1.55.0)\n",
      "Requirement already satisfied: dash>=1.5.0 in f:\\anaconda\\lib\\site-packages (from dtale) (1.21.0)\n",
      "Requirement already satisfied: flask-ngrok; python_version > \"3.0\" in f:\\anaconda\\lib\\site-packages (from dtale) (0.0.25)\n",
      "Requirement already satisfied: pandas in f:\\anaconda\\lib\\site-packages (from dtale) (1.1.3)\n",
      "Requirement already satisfied: openpyxl; python_version >= \"3.0\" in f:\\anaconda\\lib\\site-packages (from dtale) (3.0.5)\n",
      "Requirement already satisfied: strsimpy in f:\\anaconda\\lib\\site-packages (from dtale) (0.2.0)\n",
      "Requirement already satisfied: matplotlib; python_version > \"3.6\" in f:\\anaconda\\lib\\site-packages (from dtale) (3.3.2)\n",
      "Requirement already satisfied: missingno<=0.4.2 in f:\\anaconda\\lib\\site-packages (from dtale) (0.4.2)\n",
      "Requirement already satisfied: networkx; python_version >= \"3.0\" in f:\\anaconda\\lib\\site-packages (from dtale) (2.5)\n",
      "Requirement already satisfied: squarify in f:\\anaconda\\lib\\site-packages (from dtale) (0.4.3)\n",
      "Requirement already satisfied: lz4; python_version > \"3.0\" in f:\\anaconda\\lib\\site-packages (from dtale) (3.1.3)\n",
      "Requirement already satisfied: dash-colorscales in f:\\anaconda\\lib\\site-packages (from dtale) (0.0.4)\n",
      "Requirement already satisfied: scipy; python_version >= \"3.7\" in f:\\anaconda\\lib\\site-packages (from dtale) (1.5.2)\n",
      "Requirement already satisfied: kaleido; python_version > \"3.6\" in f:\\anaconda\\lib\\site-packages (from dtale) (0.2.1)\n",
      "Requirement already satisfied: scikit-learn; python_version > \"3.0\" in f:\\anaconda\\lib\\site-packages (from dtale) (0.23.2)\n",
      "Requirement already satisfied: dash-bootstrap-components; python_version > \"3.0\" in f:\\anaconda\\lib\\site-packages (from dtale) (0.13.0)\n",
      "Requirement already satisfied: xlrd in f:\\anaconda\\lib\\site-packages (from dtale) (1.2.0)\n",
      "Requirement already satisfied: Flask-Compress in f:\\anaconda\\lib\\site-packages (from dtale) (1.10.1)\n",
      "Requirement already satisfied: six in f:\\anaconda\\lib\\site-packages (from dtale) (1.15.0)\n",
      "Requirement already satisfied: statsmodels; python_version > \"3.0\" in f:\\anaconda\\lib\\site-packages (from dtale) (0.12.0)\n",
      "Requirement already satisfied: ppscore; python_version >= \"3.6\" in f:\\anaconda\\lib\\site-packages (from dtale) (1.2.0)\n",
      "Requirement already satisfied: xarray; python_version >= \"3.0\" in f:\\anaconda\\lib\\site-packages (from dtale) (0.19.0)\n",
      "Requirement already satisfied: et-xmlfile; python_version >= \"3.6\" in f:\\anaconda\\lib\\site-packages (from dtale) (1.0.1)\n",
      "Requirement already satisfied: requests in f:\\anaconda\\lib\\site-packages (from dtale) (2.24.0)\n",
      "Requirement already satisfied: itsdangerous; python_version >= \"3.7\" in f:\\anaconda\\lib\\site-packages (from dtale) (1.1.0)\n",
      "Requirement already satisfied: future>=0.14.0 in f:\\anaconda\\lib\\site-packages (from dtale) (0.18.2)\n",
      "Requirement already satisfied: Flask; python_version >= \"3.7\" in f:\\anaconda\\lib\\site-packages (from dtale) (1.1.2)\n",
      "Requirement already satisfied: dash-daq in f:\\anaconda\\lib\\site-packages (from dtale) (0.5.0)\n",
      "Requirement already satisfied: plotly>=5.0.0; python_version >= \"3.6\" in f:\\anaconda\\lib\\site-packages (from dtale) (5.2.2)\n",
      "Requirement already satisfied: seaborn; python_version >= \"3.6\" in f:\\anaconda\\lib\\site-packages (from dtale) (0.11.0)\n",
      "Requirement already satisfied: numpy; python_version >= \"3.0\" in f:\\anaconda\\lib\\site-packages (from dtale) (1.19.2)\n",
      "Requirement already satisfied: dash-html-components==1.1.4 in f:\\anaconda\\lib\\site-packages (from dash>=1.5.0->dtale) (1.1.4)\n",
      "Requirement already satisfied: dash-core-components==1.17.1 in f:\\anaconda\\lib\\site-packages (from dash>=1.5.0->dtale) (1.17.1)\n",
      "Requirement already satisfied: dash-table==4.12.0 in f:\\anaconda\\lib\\site-packages (from dash>=1.5.0->dtale) (4.12.0)\n",
      "Requirement already satisfied: python-dateutil>=2.7.3 in f:\\anaconda\\lib\\site-packages (from pandas->dtale) (2.8.1)\n",
      "Requirement already satisfied: pytz>=2017.2 in f:\\anaconda\\lib\\site-packages (from pandas->dtale) (2020.1)\n",
      "Requirement already satisfied: jdcal in f:\\anaconda\\lib\\site-packages (from openpyxl; python_version >= \"3.0\"->dtale) (1.4.1)\n",
      "Requirement already satisfied: pillow>=6.2.0 in f:\\anaconda\\lib\\site-packages (from matplotlib; python_version > \"3.6\"->dtale) (8.0.1)\n",
      "Requirement already satisfied: kiwisolver>=1.0.1 in f:\\anaconda\\lib\\site-packages (from matplotlib; python_version > \"3.6\"->dtale) (1.3.0)\n",
      "Requirement already satisfied: certifi>=2020.06.20 in f:\\anaconda\\lib\\site-packages (from matplotlib; python_version > \"3.6\"->dtale) (2020.6.20)\n",
      "Requirement already satisfied: pyparsing!=2.0.4,!=2.1.2,!=2.1.6,>=2.0.3 in f:\\anaconda\\lib\\site-packages (from matplotlib; python_version > \"3.6\"->dtale) (2.4.7)\n",
      "Requirement already satisfied: cycler>=0.10 in f:\\anaconda\\lib\\site-packages (from matplotlib; python_version > \"3.6\"->dtale) (0.10.0)\n",
      "Requirement already satisfied: decorator>=4.3.0 in f:\\anaconda\\lib\\site-packages (from networkx; python_version >= \"3.0\"->dtale) (4.4.2)\n",
      "Requirement already satisfied: threadpoolctl>=2.0.0 in f:\\anaconda\\lib\\site-packages (from scikit-learn; python_version > \"3.0\"->dtale) (2.1.0)\n",
      "Requirement already satisfied: joblib>=0.11 in f:\\anaconda\\lib\\site-packages (from scikit-learn; python_version > \"3.0\"->dtale) (0.17.0)\n",
      "Requirement already satisfied: brotli in f:\\anaconda\\lib\\site-packages (from Flask-Compress->dtale) (1.0.9)\n",
      "Requirement already satisfied: patsy>=0.5 in f:\\anaconda\\lib\\site-packages (from statsmodels; python_version > \"3.0\"->dtale) (0.5.1)\n",
      "Requirement already satisfied: setuptools>=40.4 in f:\\anaconda\\lib\\site-packages (from xarray; python_version >= \"3.0\"->dtale) (50.3.1.post20201107)\n",
      "Requirement already satisfied: urllib3!=1.25.0,!=1.25.1,<1.26,>=1.21.1 in f:\\anaconda\\lib\\site-packages (from requests->dtale) (1.25.11)\n",
      "Requirement already satisfied: chardet<4,>=3.0.2 in f:\\anaconda\\lib\\site-packages (from requests->dtale) (3.0.4)\n",
      "Requirement already satisfied: idna<3,>=2.5 in f:\\anaconda\\lib\\site-packages (from requests->dtale) (2.10)\n",
      "Requirement already satisfied: Jinja2>=2.10.1 in f:\\anaconda\\lib\\site-packages (from Flask; python_version >= \"3.7\"->dtale) (2.11.2)\n",
      "Requirement already satisfied: Werkzeug>=0.15 in f:\\anaconda\\lib\\site-packages (from Flask; python_version >= \"3.7\"->dtale) (1.0.1)\n"
     ]
    }
   ],
   "source": [
    "!pip install dtale"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Requirement already satisfied: click>=5.1 in f:\\anaconda\\lib\\site-packages (from Flask; python_version >= \"3.7\"->dtale) (7.1.2)\n",
      "Requirement already satisfied: tenacity>=6.2.0 in f:\\anaconda\\lib\\site-packages (from plotly>=5.0.0; python_version >= \"3.6\"->dtale) (8.0.1)\n",
      "Requirement already satisfied: MarkupSafe>=0.23 in f:\\anaconda\\lib\\site-packages (from Jinja2>=2.10.1->Flask; python_version >= \"3.7\"->dtale) (1.1.1)\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2021-08-24 14:18:59,077 - INFO     - NumExpr defaulting to 4 threads.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "\n",
       "        <iframe\n",
       "            width=\"100%\"\n",
       "            height=\"475\"\n",
       "            src=\"http://DESKTOP-5LSDQAR:40000/dtale/iframe/1\"\n",
       "            frameborder=\"0\"\n",
       "            allowfullscreen\n",
       "        ></iframe>\n",
       "        "
      ],
      "text/plain": [
       "<IPython.lib.display.IFrame at 0x25a3f02df10>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/plain": []
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Exception occurred while processing request: No numeric types to aggregate\n",
      "Traceback (most recent call last):\n",
      "  File \"F:\\anaconda\\lib\\site-packages\\dtale\\views.py\", line 105, in _handle_exceptions\n",
      "    return func(*args, **kwargs)\n",
      "  File \"F:\\anaconda\\lib\\site-packages\\dtale\\views.py\", line 1578, in reshape_data\n",
      "    instance = startup(data=builder.reshape(), ignore_duplicate=True)\n",
      "  File \"F:\\anaconda\\lib\\site-packages\\dtale\\data_reshapers.py\", line 46, in reshape\n",
      "    return self.builder.reshape(data)\n",
      "  File \"F:\\anaconda\\lib\\site-packages\\dtale\\data_reshapers.py\", line 60, in reshape\n",
      "    pivot_data = pd.pivot_table(\n",
      "  File \"F:\\anaconda\\lib\\site-packages\\pandas\\core\\reshape\\pivot.py\", line 112, in pivot_table\n",
      "    agged = grouped.agg(aggfunc)\n",
      "  File \"F:\\anaconda\\lib\\site-packages\\pandas\\core\\groupby\\generic.py\", line 951, in aggregate\n",
      "    result, how = self._aggregate(func, *args, **kwargs)\n",
      "  File \"F:\\anaconda\\lib\\site-packages\\pandas\\core\\base.py\", line 307, in _aggregate\n",
      "    return self._try_aggregate_string_function(arg, *args, **kwargs), None\n",
      "  File \"F:\\anaconda\\lib\\site-packages\\pandas\\core\\base.py\", line 263, in _try_aggregate_string_function\n",
      "    return f(*args, **kwargs)\n",
      "  File \"F:\\anaconda\\lib\\site-packages\\pandas\\core\\groupby\\groupby.py\", line 1393, in mean\n",
      "    return self._cython_agg_general(\n",
      "  File \"F:\\anaconda\\lib\\site-packages\\pandas\\core\\groupby\\generic.py\", line 1021, in _cython_agg_general\n",
      "    agg_blocks, agg_items = self._cython_agg_blocks(\n",
      "  File \"F:\\anaconda\\lib\\site-packages\\pandas\\core\\groupby\\generic.py\", line 1130, in _cython_agg_blocks\n",
      "    raise DataError(\"No numeric types to aggregate\")\n",
      "pandas.core.base.DataError: No numeric types to aggregate\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2021-08-24 14:20:22,719 - ERROR    - Exception occurred while processing request: No numeric types to aggregate\n",
      "Traceback (most recent call last):\n",
      "  File \"F:\\anaconda\\lib\\site-packages\\dtale\\views.py\", line 105, in _handle_exceptions\n",
      "    return func(*args, **kwargs)\n",
      "  File \"F:\\anaconda\\lib\\site-packages\\dtale\\views.py\", line 1578, in reshape_data\n",
      "    instance = startup(data=builder.reshape(), ignore_duplicate=True)\n",
      "  File \"F:\\anaconda\\lib\\site-packages\\dtale\\data_reshapers.py\", line 46, in reshape\n",
      "    return self.builder.reshape(data)\n",
      "  File \"F:\\anaconda\\lib\\site-packages\\dtale\\data_reshapers.py\", line 60, in reshape\n",
      "    pivot_data = pd.pivot_table(\n",
      "  File \"F:\\anaconda\\lib\\site-packages\\pandas\\core\\reshape\\pivot.py\", line 112, in pivot_table\n",
      "    agged = grouped.agg(aggfunc)\n",
      "  File \"F:\\anaconda\\lib\\site-packages\\pandas\\core\\groupby\\generic.py\", line 951, in aggregate\n",
      "    result, how = self._aggregate(func, *args, **kwargs)\n",
      "  File \"F:\\anaconda\\lib\\site-packages\\pandas\\core\\base.py\", line 307, in _aggregate\n",
      "    return self._try_aggregate_string_function(arg, *args, **kwargs), None\n",
      "  File \"F:\\anaconda\\lib\\site-packages\\pandas\\core\\base.py\", line 263, in _try_aggregate_string_function\n",
      "    return f(*args, **kwargs)\n",
      "  File \"F:\\anaconda\\lib\\site-packages\\pandas\\core\\groupby\\groupby.py\", line 1393, in mean\n",
      "    return self._cython_agg_general(\n",
      "  File \"F:\\anaconda\\lib\\site-packages\\pandas\\core\\groupby\\generic.py\", line 1021, in _cython_agg_general\n",
      "    agg_blocks, agg_items = self._cython_agg_blocks(\n",
      "  File \"F:\\anaconda\\lib\\site-packages\\pandas\\core\\groupby\\generic.py\", line 1130, in _cython_agg_blocks\n",
      "    raise DataError(\"No numeric types to aggregate\")\n",
      "pandas.core.base.DataError: No numeric types to aggregate\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Collecting dtale\n",
      "  Using cached dtale-1.55.0-py2.py3-none-any.whl (10.9 MB)\n",
      "Requirement already satisfied: six in f:\\anaconda\\lib\\site-packages (from dtale) (1.15.0)\n",
      "Requirement already satisfied: matplotlib; python_version > \"3.6\" in f:\\anaconda\\lib\\site-packages (from dtale) (3.3.2)\n",
      "Requirement already satisfied: scikit-learn; python_version > \"3.0\" in f:\\anaconda\\lib\\site-packages (from dtale) (0.23.2)\n",
      "Requirement already satisfied: dash>=1.5.0 in f:\\anaconda\\lib\\site-packages (from dtale) (1.21.0)\n",
      "Requirement already satisfied: future>=0.14.0 in f:\\anaconda\\lib\\site-packages (from dtale) (0.18.2)\n",
      "Collecting xarray; python_version >= \"3.0\"\n",
      "  Using cached xarray-0.19.0-py3-none-any.whl (827 kB)\n",
      "Requirement already satisfied: Flask-Compress in f:\\anaconda\\lib\\site-packages (from dtale) (1.10.1)\n",
      "Requirement already satisfied: scipy; python_version >= \"3.7\" in f:\\anaconda\\lib\\site-packages (from dtale) (1.5.2)\n",
      "Requirement already satisfied: et-xmlfile; python_version >= \"3.6\" in f:\\anaconda\\lib\\site-packages (from dtale) (1.0.1)\n",
      "Collecting plotly>=5.0.0; python_version >= \"3.6\"\n",
      "  Using cached plotly-5.2.2-py2.py3-none-any.whl (21.8 MB)\n",
      "Collecting dash-bootstrap-components; python_version > \"3.0\"\n",
      "  Using cached dash_bootstrap_components-0.13.0-py3-none-any.whl (197 kB)\n",
      "Requirement already satisfied: xlrd in f:\\anaconda\\lib\\site-packages (from dtale) (1.2.0)\n",
      "Requirement already satisfied: openpyxl; python_version >= \"3.0\" in f:\\anaconda\\lib\\site-packages (from dtale) (3.0.5)\n",
      "Requirement already satisfied: statsmodels; python_version > \"3.0\" in f:\\anaconda\\lib\\site-packages (from dtale) (0.12.0)\n",
      "Requirement already satisfied: networkx; python_version >= \"3.0\" in f:\\anaconda\\lib\\site-packages (from dtale) (2.5)\n",
      "Collecting flask-ngrok; python_version > \"3.0\"\n",
      "  Using cached flask_ngrok-0.0.25-py3-none-any.whl (3.1 kB)\n",
      "Requirement already satisfied: seaborn; python_version >= \"3.6\" in f:\\anaconda\\lib\\site-packages (from dtale) (0.11.0)\n",
      "Requirement already satisfied: requests in f:\\anaconda\\lib\\site-packages (from dtale) (2.24.0)\n",
      "Collecting lz4; python_version > \"3.0\"\n",
      "  Using cached lz4-3.1.3-cp38-cp38-win_amd64.whl (192 kB)\n",
      "Requirement already satisfied: strsimpy in f:\\anaconda\\lib\\site-packages (from dtale) (0.2.0)\n",
      "Collecting missingno<=0.4.2\n",
      "  Using cached missingno-0.4.2-py3-none-any.whl (9.7 kB)\n",
      "Requirement already satisfied: numpy; python_version >= \"3.0\" in f:\\anaconda\\lib\\site-packages (from dtale) (1.19.2)\n",
      "Collecting kaleido; python_version > \"3.6\"\n",
      "  Using cached kaleido-0.2.1-py2.py3-none-win_amd64.whl (65.9 MB)\n",
      "Processing c:\\users\\user\\appdata\\local\\pip\\cache\\wheels\\fb\\0e\\65\\da65ce669381fc01b6ef990d301d3586991f89bff1175877a4\\ppscore-1.2.0-py2.py3-none-any.whl\n",
      "Requirement already satisfied: pandas in f:\\anaconda\\lib\\site-packages (from dtale) (1.1.3)\n",
      "Processing c:\\users\\user\\appdata\\local\\pip\\cache\\wheels\\fe\\33\\33\\172dbfe1efc739352828854a113df599a428b58f68ed0c4f75\\dash_colorscales-0.0.4-py3-none-any.whl\n",
      "Requirement already satisfied: Flask; python_version >= \"3.7\" in f:\\anaconda\\lib\\site-packages (from dtale) (1.1.2)\n",
      "Requirement already satisfied: squarify in f:\\anaconda\\lib\\site-packages (from dtale) (0.4.3)\n",
      "Processing c:\\users\\user\\appdata\\local\\pip\\cache\\wheels\\00\\35\\e5\\57e90f10c529601c6627400513cb65dd5adb09752411f5a050\\dash_daq-0.5.0-py3-none-any.whl\n",
      "Requirement already satisfied: itsdangerous; python_version >= \"3.7\" in f:\\anaconda\\lib\\site-packages (from dtale) (1.1.0)\n",
      "Requirement already satisfied: certifi>=2020.06.20 in f:\\anaconda\\lib\\site-packages (from matplotlib; python_version > \"3.6\"->dtale) (2020.6.20)\n",
      "Requirement already satisfied: cycler>=0.10 in f:\\anaconda\\lib\\site-packages (from matplotlib; python_version > \"3.6\"->dtale) (0.10.0)\n",
      "Requirement already satisfied: pillow>=6.2.0 in f:\\anaconda\\lib\\site-packages (from matplotlib; python_version > \"3.6\"->dtale) (8.0.1)\n",
      "Requirement already satisfied: python-dateutil>=2.1 in f:\\anaconda\\lib\\site-packages (from matplotlib; python_version > \"3.6\"->dtale) (2.8.1)\n",
      "Requirement already satisfied: kiwisolver>=1.0.1 in f:\\anaconda\\lib\\site-packages (from matplotlib; python_version > \"3.6\"->dtale) (1.3.0)\n",
      "Requirement already satisfied: pyparsing!=2.0.4,!=2.1.2,!=2.1.6,>=2.0.3 in f:\\anaconda\\lib\\site-packages (from matplotlib; python_version > \"3.6\"->dtale) (2.4.7)\n",
      "Requirement already satisfied: joblib>=0.11 in f:\\anaconda\\lib\\site-packages (from scikit-learn; python_version > \"3.0\"->dtale) (0.17.0)\n",
      "Requirement already satisfied: threadpoolctl>=2.0.0 in f:\\anaconda\\lib\\site-packages (from scikit-learn; python_version > \"3.0\"->dtale) (2.1.0)\n",
      "Requirement already satisfied: dash-core-components==1.17.1 in f:\\anaconda\\lib\\site-packages (from dash>=1.5.0->dtale) (1.17.1)\n",
      "Requirement already satisfied: dash-table==4.12.0 in f:\\anaconda\\lib\\site-packages (from dash>=1.5.0->dtale) (4.12.0)\n",
      "Requirement already satisfied: dash-html-components==1.1.4 in f:\\anaconda\\lib\\site-packages (from dash>=1.5.0->dtale) (1.1.4)\n",
      "Requirement already satisfied: setuptools>=40.4 in f:\\anaconda\\lib\\site-packages (from xarray; python_version >= \"3.0\"->dtale) (50.3.1.post20201107)\n",
      "Requirement already satisfied: brotli in f:\\anaconda\\lib\\site-packages (from Flask-Compress->dtale) (1.0.9)\n",
      "Requirement already satisfied: tenacity>=6.2.0 in f:\\anaconda\\lib\\site-packages (from plotly>=5.0.0; python_version >= \"3.6\"->dtale) (8.0.1)\n",
      "Requirement already satisfied: jdcal in f:\\anaconda\\lib\\site-packages (from openpyxl; python_version >= \"3.0\"->dtale) (1.4.1)\n",
      "Requirement already satisfied: patsy>=0.5 in f:\\anaconda\\lib\\site-packages (from statsmodels; python_version > \"3.0\"->dtale) (0.5.1)\n",
      "Requirement already satisfied: decorator>=4.3.0 in f:\\anaconda\\lib\\site-packages (from networkx; python_version >= \"3.0\"->dtale) (4.4.2)\n",
      "Requirement already satisfied: idna<3,>=2.5 in f:\\anaconda\\lib\\site-packages (from requests->dtale) (2.10)\n",
      "Requirement already satisfied: urllib3!=1.25.0,!=1.25.1,<1.26,>=1.21.1 in f:\\anaconda\\lib\\site-packages (from requests->dtale) (1.25.11)\n",
      "Requirement already satisfied: chardet<4,>=3.0.2 in f:\\anaconda\\lib\\site-packages (from requests->dtale) (3.0.4)\n",
      "Requirement already satisfied: pytz>=2017.2 in f:\\anaconda\\lib\\site-packages (from pandas->dtale) (2020.1)\n",
      "Requirement already satisfied: Jinja2>=2.10.1 in f:\\anaconda\\lib\\site-packages (from Flask; python_version >= \"3.7\"->dtale) (2.11.2)\n",
      "Requirement already satisfied: Werkzeug>=0.15 in f:\\anaconda\\lib\\site-packages (from Flask; python_version >= \"3.7\"->dtale) (1.0.1)\n",
      "Requirement already satisfied: click>=5.1 in f:\\anaconda\\lib\\site-packages (from Flask; python_version >= \"3.7\"->dtale) (7.1.2)\n",
      "Requirement already satisfied: MarkupSafe>=0.23 in f:\\anaconda\\lib\\site-packages (from Jinja2>=2.10.1->Flask; python_version >= \"3.7\"->dtale) (1.1.1)\n",
      "Installing collected packages: xarray, plotly, dash-bootstrap-components, flask-ngrok, lz4, missingno, kaleido, ppscore, dash-colorscales, dash-daq, dtale\n",
      "  Attempting uninstall: plotly\n",
      "    Found existing installation: plotly 4.14.3\n",
      "    Uninstalling plotly-4.14.3:\n",
      "      Successfully uninstalled plotly-4.14.3\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "ERROR: Could not install packages due to an EnvironmentError: [WinError 5] Access is denied: 'F:\\\\anaconda\\\\Lib\\\\site-packages\\\\lz4\\\\_version.cp38-win_amd64.pyd'\n",
      "Consider using the `--user` option or check the permissions.\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Exception on /dtale/charts/_dash-update-component [POST]\n",
      "Traceback (most recent call last):\n",
      "  File \"F:\\anaconda\\lib\\site-packages\\flask\\app.py\", line 2447, in wsgi_app\n",
      "    response = self.full_dispatch_request()\n",
      "  File \"F:\\anaconda\\lib\\site-packages\\flask\\app.py\", line 1952, in full_dispatch_request\n",
      "    rv = self.handle_user_exception(e)\n",
      "  File \"F:\\anaconda\\lib\\site-packages\\flask\\app.py\", line 1821, in handle_user_exception\n",
      "    reraise(exc_type, exc_value, tb)\n",
      "  File \"F:\\anaconda\\lib\\site-packages\\flask\\_compat.py\", line 39, in reraise\n",
      "    raise value\n",
      "  File \"F:\\anaconda\\lib\\site-packages\\flask\\app.py\", line 1950, in full_dispatch_request\n",
      "    rv = self.dispatch_request()\n",
      "  File \"F:\\anaconda\\lib\\site-packages\\flask\\app.py\", line 1936, in dispatch_request\n",
      "    return self.view_functions[rule.endpoint](**req.view_args)\n",
      "  File \"F:\\anaconda\\lib\\site-packages\\dash\\dash.py\", line 1096, in dispatch\n",
      "    response.set_data(func(*args, outputs_list=outputs_list))\n",
      "  File \"F:\\anaconda\\lib\\site-packages\\dash\\dash.py\", line 1017, in add_context\n",
      "    output_value = func(*args, **kwargs)  # %% callback invoked %%\n",
      "  File \"F:\\anaconda\\lib\\site-packages\\dtale\\dash_application\\views.py\", line 1112, in group_values\n",
      "    group_vals = build_group_val_options(group_vals, group_cols)\n",
      "  File \"F:\\anaconda\\lib\\site-packages\\dtale\\dash_application\\layout\\layout.py\", line 1053, in build_group_val_options\n",
      "    group_vals = find_group_vals(df, group_cols)\n",
      "  File \"F:\\anaconda\\lib\\site-packages\\dtale\\charts\\utils.py\", line 1000, in find_group_vals\n",
      "    group_vals, _ = retrieve_chart_data(df, group_cols)\n",
      "  File \"F:\\anaconda\\lib\\site-packages\\dtale\\charts\\utils.py\", line 350, in retrieve_chart_data\n",
      "    all_data = pd.concat(all_data, axis=1)\n",
      "  File \"F:\\anaconda\\lib\\site-packages\\pandas\\core\\reshape\\concat.py\", line 274, in concat\n",
      "    op = _Concatenator(\n",
      "  File \"F:\\anaconda\\lib\\site-packages\\pandas\\core\\reshape\\concat.py\", line 331, in __init__\n",
      "    raise ValueError(\"No objects to concatenate\")\n",
      "ValueError: No objects to concatenate\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2021-08-24 14:23:25,098 - ERROR    - Exception on /dtale/charts/_dash-update-component [POST]\n",
      "Traceback (most recent call last):\n",
      "  File \"F:\\anaconda\\lib\\site-packages\\flask\\app.py\", line 2447, in wsgi_app\n",
      "    response = self.full_dispatch_request()\n",
      "  File \"F:\\anaconda\\lib\\site-packages\\flask\\app.py\", line 1952, in full_dispatch_request\n",
      "    rv = self.handle_user_exception(e)\n",
      "  File \"F:\\anaconda\\lib\\site-packages\\flask\\app.py\", line 1821, in handle_user_exception\n",
      "    reraise(exc_type, exc_value, tb)\n",
      "  File \"F:\\anaconda\\lib\\site-packages\\flask\\_compat.py\", line 39, in reraise\n",
      "    raise value\n",
      "  File \"F:\\anaconda\\lib\\site-packages\\flask\\app.py\", line 1950, in full_dispatch_request\n",
      "    rv = self.dispatch_request()\n",
      "  File \"F:\\anaconda\\lib\\site-packages\\flask\\app.py\", line 1936, in dispatch_request\n",
      "    return self.view_functions[rule.endpoint](**req.view_args)\n",
      "  File \"F:\\anaconda\\lib\\site-packages\\dash\\dash.py\", line 1096, in dispatch\n",
      "    response.set_data(func(*args, outputs_list=outputs_list))\n",
      "  File \"F:\\anaconda\\lib\\site-packages\\dash\\dash.py\", line 1017, in add_context\n",
      "    output_value = func(*args, **kwargs)  # %% callback invoked %%\n",
      "  File \"F:\\anaconda\\lib\\site-packages\\dtale\\dash_application\\views.py\", line 1112, in group_values\n",
      "    group_vals = build_group_val_options(group_vals, group_cols)\n",
      "  File \"F:\\anaconda\\lib\\site-packages\\dtale\\dash_application\\layout\\layout.py\", line 1053, in build_group_val_options\n",
      "    group_vals = find_group_vals(df, group_cols)\n",
      "  File \"F:\\anaconda\\lib\\site-packages\\dtale\\charts\\utils.py\", line 1000, in find_group_vals\n",
      "    group_vals, _ = retrieve_chart_data(df, group_cols)\n",
      "  File \"F:\\anaconda\\lib\\site-packages\\dtale\\charts\\utils.py\", line 350, in retrieve_chart_data\n",
      "    all_data = pd.concat(all_data, axis=1)\n",
      "  File \"F:\\anaconda\\lib\\site-packages\\pandas\\core\\reshape\\concat.py\", line 274, in concat\n",
      "    op = _Concatenator(\n",
      "  File \"F:\\anaconda\\lib\\site-packages\\pandas\\core\\reshape\\concat.py\", line 331, in __init__\n",
      "    raise ValueError(\"No objects to concatenate\")\n",
      "ValueError: No objects to concatenate\n"
     ]
    }
   ],
   "source": [
    "import dtale\n",
    "dtale.show(df)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.5"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
